src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -852,6 +936,9 @@ void Terminal::_AdjustCursorPosition(const COORD proposedPosition)
             proposedCursorPosition.Y--;
             rowsPushedOffTopOfBuffer++;
         }
+
+        // manually erase our pattern intervals since the locations have changed now
+        _patternIntervalTree = {};

```

Comment:
Flagging this line in particular for @miniksa and @DHowett 

This is for the situation where we start scrolling the buffer underneath the viewport - the old pattern interval tree needs to be thrown out but Control doesn't know about it so we do it in Core. Control will still tell the Core to get a new pattern interval tree since presumably there was output to get us here.

Reply:
That's fine. If it works, it works. With a bunch more mental gymnastics, we could probably make this more sensible... but I don't really want to hold this PR up for it.